### PR TITLE
feat: allow usage of collateral permits for minting

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
To allow for collateral permits, I decided to split minting into two separate functions. The rationale for this is that some contracts might want to mint positions (e.g. ArbitrageBroker) and contracts wouldn't process permits on their own behalf. Thus the simplest solution was to override the function and create a separate mint function with an additional permit parameter. This also means existing contracts and tests do not have to modified further to accomodate this change.

### CODE REVIEW

**Comment by ramenforbreakfast** [comment]
Updated to use `permitAndMint` function signature rather than overridden `mint` function

File: apps/smart-contracts/core/contracts/PrePOMarket.sol [hunk 1]
```diff
@@ -117,24 +117,28 @@ contract PrePOMarket is
...
-    _longToken.mint(recipient, longShortAmount);
-    _shortToken.mint(recipient, longShortAmount);
-    emit Mint(msg.sender, recipient, longShortAmount);
+    return _mint(collateralAmount, recipient);
+  }
+
+  function permitAndMint(
+    uint256 collateralAmount,
+    address recipient,
+    Permit calldata permit
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/contracts/PrePOMarket.sol:128]: nit: put this arg first to match better with the function name

File: apps/smart-contracts/core/test/unit/PrePOMarket.test.ts [hunk 2]
```diff
@@ -1058,6 +1072,90 @@ describe('=> prePOMarket', () => {
...
+      expect(await collateralToken.balanceOf(user1.address)).eq(userCTBefore.sub(TEST_MINT_AMOUNT))
+      expect(await collateralToken.balanceOf(governance.address)).eq(
+        treasuryCTBefore.add(expectedMintingFeeAmount)
+      )
+      expect(await collateralToken.balanceOf(prePOMarket.address)).eq(
+        marketCTBefore.add(TEST_MINT_AMOUNT.sub(expectedMintingFeeAmount))
+      )
+    })
+
+    it('ignores collateral permit if deadline == 0', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/PrePOMarket.test.ts:1132]:
```suggestion
    it('ignores collateral permit if deadline = 0', async () => {
```

File: apps/smart-contracts/core/test/unit/PrePOMarket.test.ts [hunk 3]
```diff
@@ -1058,6 +1072,90 @@ describe('=> prePOMarket', () => {
...
+      expect(await collateralToken.allowance(user1.address, prePOMarket.address)).eq(0)
+      const userCTBefore = await collateralToken.balanceOf(user1.address)
+      const treasuryCTBefore = await collateralToken.balanceOf(governance.address)
+      const marketCTBefore = await collateralToken.balanceOf(prePOMarket.address)
+
+      await prePOMarket
+        .connect(user1)
+        .permitAndMint(TEST_MINT_AMOUNT, user2.address, permitForFunder)
+
+      expect(await collateralToken.allowance(user1.address, prePOMarket.address)).eq(
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/PrePOMarket.test.ts:1118]: add expect that permit was called with correct arguments

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

**Pull request merged**